### DISCUSSION TITLE
Failing with database issue

### DISCUSSION THREAD

**Discussion by dwchamberlin** [bourquep/weewx-me.teo#60] [Q&A]
I tried installing the skin on my setup because it looked very close to what I want for one of my displays.  However when it tries to generate the report, I get a string of errors:

`Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: Evaluation of template /etc/weewx/skins/me.teo/data/day-%Y-%m-%d.json.tmpl failed with exception '<class 'sqlite3.DatabaseE
rror'>'                                                                                                                                                                                          
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/me.teo/data/day-%Y-%m-%d.json.tmpl                                                 
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: **** Reason: database disk image is malformed                                                                              
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****  Traceback (most recent call last):                                                                                   
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/cheetahgenerator.py", line 334, in generate                                           
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****      unicode_string = compiled_template.respond()                                                                     
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^                                                                     
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "_etc_weewx_skins_me_teo_data_day__Y__m__d_json_tmpl.py", line 203, in respond                                
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/tags.py", line 415, in series                                                         
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****      start_vt, stop_vt, data_vt = weewx.xtypes.get_series(
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****                                   ^^^^^^^^^^^^^^^^^^^^^^^^
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/xtypes.py", line 109, in get_series
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****      return xtype.get_series(obs_type, timespan, db_manager, aggregate_type,
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/xtypes.py", line 242, in get_series
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****      for record in db_manager.genSql(sql_str, (startstamp, stopstamp)):
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****    File "/usr/share/weewx/weewx/manager.py", line 618, in genSql
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****      for _row in _cursor.execute(sql, sqlargs):
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: ****  sqlite3.DatabaseError: database disk image is malformed
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: Evaluation of template /etc/weewx/skins/me.teo/data/day-%Y-%m-%d.json.tmpl failed with exception '<class 'sqlite3.DatabaseE
rror'>'                                                                         
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: **** Ignoring template /etc/weewx/skins/me.teo/data/day-%Y-%m-%d.json.tmpl
Dec 01 16:14:20 hampi5 weewxd[3574824]: ERROR weewx.cheetahgenerator: **** Reason: database disk image is malformed
`

And a bunch more similar complaints.  It seems not to like my database.  I'm running weewx 5.1.0.  I did recently upgrade from an older weewx (v4.1.1) but I went through a `database reconfigure` and `database rebuild-daily`.  All of my other skins are running fine.  

And in a side-note, I'd also vote for adding MQTT.  My main weather page uses the belchertown skin which uses MQTT and I really like it ( https://pacificaview.net/weather/ )

**Comment by bourquep** [comment]
Hello,

The skin itself does not interact directly with the database. It uses well-documented template conventions, and the Cheetah generator takes care of fetching data from the db to process those templates.

In this case, it's the daily data template `day-%Y-%m-%d.json.tmpl` that fails. 

Does the rest of the files get generated properly despite those errors?

**Comment by bourquep** [comment]
@dwchamberlin Were you able to figure this out?
